site-service/src/test/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceTests.java

```diff
@@ -0,0 +1,126 @@
+        when(siteStore.saveEntity(any(Site.class))).thenReturn("123");
+        //Act
+        siteService.saveSite(siteWithParent);
+
+        //Assert
+        Mockito.verify(siteValidation).validate(siteCaptor.capture());
+        var value = siteCaptor.getValue();
+        assertThat(siteWithParent, equalTo(value));
+    }
+
+    @Test
+    void TestSaveSite_WithSite_ValidatesSite() {
+        // Arrange
+        String name = "name";
+        String alias = "alias";
+        Site site = new Site(name, alias);
+        var siteService = new SiteService(siteStore, siteValidation);
+        when(siteValidation.validate(any(Site.class))).thenReturn(new ValidationResponse(true, ""));
+        when(siteStore.saveEntity(any(Site.class))).thenReturn("123");
+        //Act
+        siteService.saveSite(site);
+
+        //Assert
+        Mockito.verify(siteValidation).validate(siteCaptor.capture());
+        var value = siteCaptor.getValue();
+        assertThat(site, equalTo(value));
+    }
+
+    @Test
+    void TestSaveSite_WhenValidSite_SavesToStore(){
+        // Arrange
+        String name = "name";
+        String alias = "alias";
+        Site site = new Site(name, alias);
+        var siteService = new SiteService(siteStore, siteValidation);
+        when(siteValidation.validate(any(Site.class))).thenReturn(new ValidationResponse(true, ""));
+        when(siteStore.saveEntity(any(Site.class))).thenReturn("123");
+        //Act
+        siteService.saveSite(site);
+
+        //Assert
+        Mockito.verify(siteStore).saveEntity(siteCaptor.capture());
+        var value = siteCaptor.getValue();
+        assertThat(site, equalTo(value));
+    }
+
+    @Test
+    void TestSaveSite_WhenInvalidSite_ThrowsValidationException(){
+        // Arrange
+        String name = "name";
+        String alias = "alias";
+        Site site = new Site(name, alias);
+        var siteService = new SiteService(siteStore, siteValidation);
+        when(siteValidation.validate(any(Site.class))).thenReturn(new ValidationResponse(false, "name"));
+        //Act + Assert
+        Assertions.assertThrows(ValidationException.class, () -> siteService.saveSite(site),
+                "Expecting save to throw validation exception");
+    }
+
+    @Test
+    void TestSaveSite_WhenInvalidSite_DoesntSavesToStore(){
+        // Arrange
+        String name = "name";
+        String alias = "alias";
+        Site site = new Site(name, alias);
+        var siteService = new SiteService(siteStore, siteValidation);
+        when(siteValidation.validate(any(Site.class))).thenReturn(new ValidationResponse(false, "name"));
+        //Act + Assert
+        Assertions.assertThrows(ValidationException.class, () -> siteService.saveSite(site),
+                "Expecting save to throw validation exception");
+        Mockito.verify(siteStore, Mockito.times(0)).saveEntity(any(Site.class));
```

Comment:
This is the only added line that doesn't exist in the previous test. I think we can merge the two.

Reply:
Merged